server/service/src/apis/api_on_central/mod.rs

```diff
@@ -0,0 +1,122 @@
+use log::info;
+use repository::{NameStoreJoinRepository, NameStoreJoinRow, RepositoryError};
+use reqwest::{ClientBuilder, StatusCode};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+use url::Url;
+use util::format_error;
+
+use crate::{
+    service_provider::{ServiceContext, ServiceProvider},
+    sync::CentralServerConfig,
+};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NameStoreJoinParams {
+    pub id: String,
+    pub name_id: String,
+    pub store_id: String,
+}
+
+#[derive(Debug)]
+pub struct SiteAuth {
+    pub username: String,
+    pub password_sha256: String,
+}
+
+#[derive(Deserialize, Debug, Error, Serialize)]
+pub enum CentralApiError {
+    #[error("Not a central server")]
+    NotACentralServer,
+    #[error("Not authorized")]
+    NotAuthorized,
+    #[error("Connection error: {0}")]
+    ConnectionError(String),
+    #[error("Internal error: {0}")]
+    InternalError(String),
+}
+
+impl From<RepositoryError> for CentralApiError {
+    fn from(from: RepositoryError) -> Self {
+        CentralApiError::InternalError(format_error(&from))
+    }
+}
+
+/// Creates/updates a name_store_join for a patient
+pub async fn patient_name_store_join(
+    service_provider: &ServiceProvider,
+    NameStoreJoinParams {
+        id,
+        name_id,
+        store_id,
+    }: NameStoreJoinParams,
+    auth: SiteAuth,
+) -> Result<(), CentralApiError> {
+    if !CentralServerConfig::is_central_server() {
+        return Err(CentralApiError::NotACentralServer);
+    }
+
+    let ctx = service_provider.basic_context()?;
+    validate_auth(service_provider, &ctx, &auth).await?;
+
+    let name_store_join_repo = NameStoreJoinRepository::new(&ctx.connection);
+
+    // TODO: maybe should prevent this from creating a changelog? Let the OG one be source of truth?
```

Comment:
I'm guessing no major harm - do we have a "special case plz don't make changelog" yet? Just some extra sync records, and mayyyybe could overwrite something on OG, but seems very low likelihood

Reply:
[f7108a7](https://github.com/msupply-foundation/open-msupply/pull/7161/commits/f7108a7be4ccddb3007cf38ce963891cc12ca125)